### DISCUSSION TITLE
Unit Testing Stack with SSM Lookup

### DISCUSSION THREAD

**Discussion by miles-po** [aws/aws-cdk#26743] [Q&A]
What are best practices for unit testing when your stack includes SSM Parameter Store lookups? I've tried mocking StringParameter.valueFromLookup, but the patterns/examples from the Jest site don't seem to work. The test always fails with an error about account and region being missing, implying that the function is not in fact mocked.

How does one mock SSM lookups or avoid them in the unit tests by other means?

**Comment by larskinder** [comment]
Hi, 

I had a similar issue today, having to lookup a JSON object, but kept getting back the dummy string, leading to JSON parse failing.

Anyway, you can mock ssm lookups by setting the app context to include your ssm parameters.

E.g. 
```
describe('pipeline-stack-test-feature', () => {
  const app = new cdk.App({
    context: {
      'ssm:account=123456789012:parameterName=/test/test/api_gw_endpoint:region=eu-west-1': '{"apiEndpoint":"https://1234567.execute-api.eu-west-1.amazonaws.com/test/"}',
    },
  });
  ```
